SpeedyTransforms/src/gradient_arrays.jl

```diff
@@ -0,0 +1,177 @@
+"""$(TYPEDSIGNATURES)
+Recursion factors `ϵ` as a function of degree `l` and order `m` (0-based) of the spherical harmonics.
+ϵ(l, m) = sqrt((l^2-m^2)/(4*l^2-1))."""
+recursion_factor(l, m) = sqrt((l^2 - m^2) / (4 * l^2 - 1))
+
+"""$(TYPEDSIGNATURES)     
+Returns a matrix of recursion factors `ϵ` up to degree `lmax` and order `mmax` (1-based) of the `spectrum` in number format `NF`."""
+function recursion_factors(
+        ::Type{NF},             # number format NF
+        spectrum::Spectrum,
+    ) where {NF}
+
+    ϵ_lm = zeros(NF, spectrum)                              # store in lower triangular matrix
```

Comment:
So this is currently only working with CPU spectra because of the loop below. 

So far what we did is just to do these pre computations on CPU and then the constructor of the struct does an automatic conversion in the end. 

This is done below in the `spectral_transforms.jl` where we are specifically declaring a `spectrum_cpu` just for initialisation purposes. 

We could also write kernels for all of these functions, but it's only at initialisation and it's all rather small matrices, so I wouldn't see this as very urgent.

Reply:
Without looking at it, I also assume that's why the GPU CI is failing.